pkg/morph/client/multi.go

```diff
@@ -120,6 +132,79 @@ func (c *Client) notificationLoop() {
 	}
 }
 
+func (c *Client) switchToMostPrioritized() {
+	const period = 30 * time.Second
+
+	t := time.NewTicker(period)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-c.cfg.ctx.Done():
+			return
+		case <-t.C:
+			c.switchLock.RLock()
+			currPriority := c.endpoints.list[c.endpoints.curr].Priority
+
+			var tryE string
+			for i, e := range c.endpoints.list {
+				if i == c.endpoints.curr {
+					continue
+				}
+
+				if e.Priority < currPriority {
+					tryE = e.Address
+					break
+				}
+			}
```

Comment:
Endpoints are already sorted by priority, so we can just try each node in list until the current one.
This also helps in a situation when we failed to connect to the most prioritized node: in this case the next should be tried in the same loop.

Reply:
> Endpoints are already sorted by priority, so we can just try each node in list until the current one.

If we have `[a, b, c, d]` nodes with `[1, 2, 2, 2]` priorities and are connected to `d`, it is useless for us to reconnect to any node that is not `a`. At least this is how i understand the priorities

> we failed to connect to the most prioritized node: in this case the next should be tried in the same loop

well, that is one of the questions i was not sure about. ok, added that